kube-client/src/client/auth/mod.rs

```diff
@@ -206,10 +206,11 @@ impl RefreshableToken {
             }
 
             RefreshableToken::File(token_file) => {
-                if let Some(token) = token_file.read().await.cached_token() {
+                let mut locked = token_file.lock().await;
+                if let Some(token) = locked.cached_token() {
                     bearer_header(token)
                 } else {
-                    bearer_header(token_file.write().await.token())
+                    bearer_header(locked.token())
                 }
```

Comment:
So, this can lead to the deadlock explained in the docs?

> Note that under the priority policy of [`RwLock`](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html), read locks are not granted until prior write locks, to prevent starvation. Therefore deadlock may occur if a read lock is held by the current task, a write lock attempt is made, and then a subsequent read lock attempt is made by the current task.
>
> Returns an RAII guard which will drop this read access of the `RwLock` when dropped.

I thought the `RwLockReadGuard` from `.read().await` is dropped before the `else` block :/

Reply:
So did I, to be honest.. :/